commit c40f44ef9b35
Author: Xidorn Quan <me@upsuper.org>
Date:   Wed Aug 9 19:57:33 2017 +1000

    Bug 1370508 - Enable stylo on generic XML documents. r=heycam
    
    MozReview-Commit-ID: 4XQ9RPQ7oa1
    
    --HG--
    extra : rebase_source : 74e411e2280e818fc1fbd2efda677565cfa2f8b6
---
 dom/base/nsDocument.cpp                                           | 3 +--
 layout/reftests/w3c-css/failures.list                             | 4 ++++
 layout/reftests/w3c-css/received/reftest.list                     | 2 +-
 layout/xul/crashtests/crashtests.list                             | 2 +-
 testing/web-platform/meta/css/css-namespaces-3/syntax-013.xml.ini | 4 ++++
 5 files changed, 11 insertions(+), 4 deletions(-)

diff --git dom/base/nsDocument.cpp dom/base/nsDocument.cpp
index 2a01e6ec3a67..66a860ff3257 100644
--- dom/base/nsDocument.cpp
+++ dom/base/nsDocument.cpp
@@ -13293,8 +13293,7 @@ nsIDocument::UpdateStyleBackendType()
       mStyleBackendType = StyleBackendType::Servo;
     } else if (!mDocumentContainer) {
       NS_WARNING("stylo: No docshell yet, assuming Gecko style system");
-    } else if ((IsHTMLOrXHTML() || IsSVGDocument()) &&
-               IsContentDocument()) {
+    } else if (!IsXULDocument() && IsContentDocument()) {
       // Disable stylo for about: pages other than about:blank, since
       // they tend to use unsupported selectors like XUL tree pseudos.
       bool isAbout = false;
diff --git layout/reftests/w3c-css/failures.list layout/reftests/w3c-css/failures.list
index 966f69ff9ae0..1560f4ad673d 100644
--- layout/reftests/w3c-css/failures.list
+++ layout/reftests/w3c-css/failures.list
@@ -314,3 +314,7 @@ fuzzy(255,2808) css-multicol-1/multicol-rule-large-001.xht
 fails-if(!styloVsGecko) css-multicol-1/multicol-fill-auto-block-children-001.xht
 fails-if(!styloVsGecko) css-multicol-1/multicol-fill-auto-block-children-002.xht
 fails-if(!styloVsGecko) css-multicol-1/multicol-span-all-block-sibling-003.xht
+
+#### CSS Namespaces 3 ##############################################
+
+fails-if(stylo||styloVsGecko) css-namespaces-3/syntax-013.xml
diff --git layout/reftests/w3c-css/received/reftest.list layout/reftests/w3c-css/received/reftest.list
index eb6151266acd..128d1fcf8121 100644
--- layout/reftests/w3c-css/received/reftest.list
+++ layout/reftests/w3c-css/received/reftest.list
@@ -205,7 +205,7 @@ fails-if(!styloVsGecko) == css-multicol-1/multicol-zero-height-001.xht css-multi
 == css-namespaces-3/syntax-010.xml css-namespaces-3/reftest/ref-lime-3.xml
 == css-namespaces-3/syntax-011.xml css-namespaces-3/reftest/ref-lime-6.xml
 == css-namespaces-3/syntax-012.xml css-namespaces-3/reftest/ref-lime-3.xml
-== css-namespaces-3/syntax-013.xml css-namespaces-3/reftest/ref-lime-5.xml
+fails-if(stylo||styloVsGecko) == css-namespaces-3/syntax-013.xml css-namespaces-3/reftest/ref-lime-5.xml # bug 1388911
 == css-namespaces-3/syntax-014.xml css-namespaces-3/reftest/ref-lime-3.xml
 == css-namespaces-3/syntax-015.xml css-namespaces-3/reftest/ref-lime-1.xml
 fails-if(!styloVsGecko) == css-values-3/attr-color-invalid-cast.html css-values-3/reference/200-200-green.html
diff --git layout/xul/crashtests/crashtests.list layout/xul/crashtests/crashtests.list
index c1445cb5ec7a..e60af3526383 100644
--- layout/xul/crashtests/crashtests.list
+++ layout/xul/crashtests/crashtests.list
@@ -1,6 +1,6 @@
 load 131008-1.xul
 load 137216-1.xul
-load 140218-1.xml
+asserts-if(stylo,3) load 140218-1.xml
 load 151826-1.xul
 load 168724-1.xul
 load 189814-1.xul
diff --git testing/web-platform/meta/css/css-namespaces-3/syntax-013.xml.ini testing/web-platform/meta/css/css-namespaces-3/syntax-013.xml.ini
new file mode 100644
index 000000000000..b034ccf978f5
--- /dev/null
+++ testing/web-platform/meta/css/css-namespaces-3/syntax-013.xml.ini
@@ -0,0 +1,4 @@
+[syntax-013.xml]
+  type: reftest
+  expected:
+    if stylo: FAIL
